lib/SILGen/SILGenThunk.cpp

```diff
@@ -310,7 +310,7 @@ SILGenModule::getOrCreateForeignAsyncCompletionHandlerImplFunction(
           // Convert the ObjC argument to the bridged Swift representation we
           // want.
           ManagedValue bridgedArg = SGF.emitBridgedToNativeValue(loc,
-                                       arg,
+                                       arg.copy(SGF, loc),
```

Comment:
My suggestion: if you put in an ensure plus one here then you won't copy if the value is already plus one.

Reply:
I had tried that, but apparently `arg` is already +1 here. The issue is that if `arg` gets consumed directly on the success path, then it's left unbalanced on the error path where it isn't used. Copying on the success branch here seems to me like the most straightforward way to keep both branches balanced without some more complicated cleanup tracking and resetting. I figured that, if optimization passes can clean up the extra copy after SILGen, it's on balance better to keep SILGen simpler here.